projects/storefrontlib/router/on-navigate.service.spec.ts

```diff
@@ -91,14 +99,37 @@ describe('OnNavigateService', () => {
       expect(viewportScroller.scrollToPosition).toHaveBeenCalledWith([0, 0]);
     });
 
+    it('should NOT scroll to the top on navigation when route has query strings', () => {
+      config.enableResetViewOnNavigate.ignoreQueryString = true;
+
+      service.setResetViewOnNavigate(true);
+
+      emitPairScrollEvent(null, '/test2?spartacus=true', '/test2');
+
+      expect(viewportScroller.scrollToPosition).not.toHaveBeenCalledWith([
+        0, 0,
+      ]);
+    });
+
+    it('should NOT scroll to the top on navigation when route is a child route', () => {
```

Comment:
Could you add a test case where you test the scroll (like the test bellow) but with ignore routes configured.